ergo/caip2.md

```diff
@@ -0,0 +1,77 @@
+---
+namespace-identifier: ergo-caip2
+title: Ergo Blockchain ID Specification
+author: Yuriy Gagarin (@gagarin55)
+discussions-to: https://github.com/ChainAgnostic/namespaces/pull/98
+status: Draft
+type: Standard
+created: 2023-11-02
+updated: 2023-11-09
+---
+
+
+## Abstract
+
+In CAIP-2 a general blockchain identification scheme is defined. This is the
+implementation of CAIP-2 for Ergo network.
+
+
+## Specification
+
+Blockchains in the "ergo" namespace are identified by their chain ID.
+
+Chain ID is 32-character prefix from the hash of the genesis block of a given chain, in lowercase hex representation.
+
+
+### Syntax
+
+The `chain_id` is a case-sensitive string in the form
+
+```
+chain_id:    namespace + ":" + reference
+namespace:   ergo
+reference:   32-character prefix from the hash of the genesis block
```

Comment:
Why 32 characters? If you're already substringing out a prefix, you might be using up valuable characters that you could need for the address types (CAIP-10, CAIP-19) and possible paths or query parameters tacked on at the end. If you check out other CAIP-2 profiles that substring a hash, they tend to opt for 8 characters or even 4, if they're not anticipating too much collision-risk between hundreds of thousands of private networks and testnets :D

Reply:
(This is not to say 8 or 4 is optimal, just recommending you add a small `### Rationale` section if there is, in fact, a meaning to using a 32char reference! )

Reply:
also, if there are only 16 possible chainIDs in the prefixes (half an 8bit unsigned --> `0`-`15`, right?), chainId could also just that prefix?  just a thought, no strong preference here just pointing out that `0` and `15` are both equally good values here, if you want to economize or use them as a checksum against the prefix in the CAIP-10/19 reference that follows!

Reply:
I borrowed the concept of using a 32-character identifier from another blockchain's CAIP-2.

Allow me to clarify the concept. In Ergo addresses, we should view the `network type prefix` not merely as a specific chain ID, but rather as a designation of the `blockchain type`—whether it's a testnet, mainnet, etc.
For example, if we were to fork the mainnet, we would create another chain with the same `blockchain type` but a different chain ID (similar to Bitcoin forks).
Alternatively, consider a scenario where someone decides to launch a new testnet with a new genesis block but the same address scheme.

I am not an Ergo core developer, so the current situation stands as described.